src/hooks/useDomainSearch.tsx

```diff
@@ -0,0 +1,54 @@
+import type { Domain } from '@zero-tech/zns-sdk';
+
+import { useState } from 'react';
+import { useQuery } from 'react-query';
+import { useZnsSdk } from './useZnsSdk';
+import { getDomainId } from '../helpers/domains';
+import { MIN_DOMAIN_NAME_LENGTH } from '../constants/domains';
+
+export interface UseDomainSearchReturn {
+	isLoading: boolean;
+	pattern: string;
+	setPattern: (pattern: string) => void;
+	exactMatch?: Domain;
+}
+
+export const useDomainSearch = (): UseDomainSearchReturn => {
```

Comment:
It could be nice abstracting some of this into a hook that only checks domain availability. For example, `useDomainAvailability`, which returns `isAvailable`. This is a little bit nicer, because you won't have to handle the logic of `isAvailable = exactMatch === undefined` in the component using the hook.